### PR TITLE
feat: Add `/suggestSource` endpoint for external use

### PR DESCRIPTION
**Pull request by kouloumos**
Refactors source suggestion logic from `useGitHub` hook into a new `/suggestSource` endpoint, enabling its use by external applications like the Registry website. Implements GitHub App authentication as a fallback for non-session-based requests, ensuring the "suggest a source for transcription" workflow can be completed by unauthenticated users from external apps.

**Notes:**

- This is the command that I am using to get the base64 which goes in the `.env` file as `GITHUB_CURATOR_PRIVATE_KEY_BASE64`
  ```
  base64 -w 0 your_private_key.pem > your_private_key_base64.txt
  ```
- In order to **test this**, you will need valid values for the newly added env variables (see `.env.local.example`) in order for the GitHub App to function.

### CODE REVIEW

File: src/pages/api/github/newBranch.ts [hunk 1]
```diff
@@ -1,10 +1,10 @@
 import { upstreamOwner } from "@/config/default";
-import { Octokit } from "@octokit/core";
+import { Octokit } from "@octokit/rest";
```
Comment (Emmanuel-Develops) [inline src/pages/api/github/newBranch.ts:2]: is this library backwards compatible with `@octokit/core`?

**Comment by Emmanuel-Develops** [comment]
> I'm not entirely certain if this is the optimal approach for handling the private key. I would appreciate any feedback or suggestions on this implementation.

Having the private key file in the deployed environment will be difficult, if at all possible, without writing to the fileSystem at build time.

What I've done however is convert the file to a base64 encoded string. 
I worked with a `.pem` file, converted to `base64`, and then saved that string as an env variable `PRIVATE_KEY`.
When you want to read the file you can then do
```js 
const privateKey = Buffer.from(process.env.PRIVATE_KEY, "base64")
```

File: src/pages/api/github/newBranch.ts (hunk 1 above)
Comment (kouloumos) [reply src/pages/api/github/newBranch.ts:2]: Yes, the `@octokit/rest` library extends `@octokit/core` by adding methods for common GitHub REST API endpoints, while @octokit/core is a lower-level, more flexible client.

 `@octokit/rest` was needed for the GitHub App authentication, and then I had type issues with the rest of octokit usage, that's why I also switched to that in the rest of the files.

**Comment by kouloumos** [comment]
@Emmanuel-Develops I [updated the PR](https://github.com/bitcointranscripts/transcription-review-front-end/compare/6b0d5ae3fac9021adb1cbabde9b688ff68cbd007..97506491b093d9ef72e2308ceece5067584b3aa0) to use base64 encoding as well as the PR description.

**Comment by kouloumos** [comment]
@Emmanuel-Develops pointed out that GitHub app authentication shouldn't' be the default fallback for all github routes.

I [forced pushed the PR](https://github.com/bitcointranscripts/transcription-review-front-end/compare/97506491b093d9ef72e2308ceece5067584b3aa0..e301c48b4bc1cf6e5a71a9e5d032d65782b1677b) (and [again](https://github.com/bitcointranscripts/transcription-review-front-end/compare/e301c48b4bc1cf6e5a71a9e5d032d65782b1677b..0508643f5d6df6e2fc64b1eab8e7f4dc481fe8ef)) to fix that by adding a `allowAppFallback` flag to the `getOctokit` method. 

To avoid the need for repeating the error handling logic across all github routes, I also added an error handler for consistency across the related routes.

I have tested that all the github routes are working.

**Review by Emmanuel-Develops — APPROVED** [review]

**Pull request merged**